netbox/dcim/api/views.py

```diff
@@ -371,8 +372,6 @@ def napalm(self, request, pk):
         Execute a NAPALM method on a Device
         """
         device = get_object_or_404(Device, pk=pk)
-        if not device.primary_ip:
```

Comment:
Device names are optional, so we still want to ensure that a device has a primary IP _or_ name defined.

Reply:
👍  Updated to have a check for device.name and raise exception if not found.

Reply:
We want to keep this check at the beginning of the method, both to keep it grouped with the other checks and to ensure that failures occur quickly. (There's no point in initializing the NAPALM driver only to then discover that we have no way of connecting to the device.)